diff --git a/feature/ssh/ssh.go b/feature/ssh/ssh.go
index bd22005..6471811 100644
--- a/feature/ssh/ssh.go
+++ b/feature/ssh/ssh.go
@@ -1,7 +1,7 @@
 // Copyright (c) Tailscale Inc & contributors
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build ((linux && !android) || (darwin && !ios) || freebsd || openbsd || plan9) && !ts_omit_ssh
+//go:build (linux || (darwin && !ios) || freebsd || openbsd || plan9) && !ts_omit_ssh
 
 // Package ssh registers the Tailscale SSH feature, including host key
 // management and the SSH server.
diff --git a/net/netmon/netmon_linux.go b/net/netmon/netmon_linux.go
index b7d87f9..317570d 100644
--- a/net/netmon/netmon_linux.go
+++ b/net/netmon/netmon_linux.go
@@ -1,8 +1,6 @@
 // Copyright (c) Tailscale Inc & contributors
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build !android
-
 package netmon
 
 import (
diff --git a/net/netmon/netmon_polling.go b/net/netmon/netmon_polling.go
index bdeb430..c5f7989 100644
--- a/net/netmon/netmon_polling.go
+++ b/net/netmon/netmon_polling.go
@@ -1,7 +1,7 @@
 // Copyright (c) Tailscale Inc & contributors
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build (!linux && !freebsd && !windows && !darwin) || android
+//go:build !linux && !freebsd && !windows && !darwin
 
 package netmon
 
diff --git a/net/routetable/routetable_linux.go b/net/routetable/routetable_linux.go
index 479aa8f..6337b61 100644
--- a/net/routetable/routetable_linux.go
+++ b/net/routetable/routetable_linux.go
@@ -1,7 +1,7 @@
 // Copyright (c) Tailscale Inc & contributors
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build linux && !android
+//go:build linux
 
 package routetable
 
diff --git a/net/routetable/routetable_other.go b/net/routetable/routetable_other.go
index 25d008c..64cd9be 100644
--- a/net/routetable/routetable_other.go
+++ b/net/routetable/routetable_other.go
@@ -1,7 +1,7 @@
 // Copyright (c) Tailscale Inc & contributors
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build android || (!linux && !darwin && !freebsd && !openbsd)
+//go:build !linux && !darwin && !freebsd && !openbsd
 
 package routetable
 
diff --git a/net/tsaddr/tsaddr.go b/net/tsaddr/tsaddr.go
index 1eac9eb..b79d679 100644
--- a/net/tsaddr/tsaddr.go
+++ b/net/tsaddr/tsaddr.go
@@ -8,6 +8,7 @@
 	"encoding/binary"
 	"errors"
 	"net/netip"
+	"os"
 	"slices"
 	"sync"
 
@@ -26,12 +27,19 @@ func ChromeOSVMRange() netip.Prefix {
 
 var chromeOSRange oncePrefix
 
+func cgnatRangeCIDR() string {
+	if v := os.Getenv("TS_RANGE"); v != "" {
+		return v
+	}
+	return "100.64.0.0/10"
+}
+
 // CGNATRange returns the Carrier Grade NAT address range that
 // is the superset range that Tailscale assigns out of.
 // See https://tailscale.com/s/cgnat
 // Note that Tailscale does not assign out of the ChromeOSVMRange.
 func CGNATRange() netip.Prefix {
-	cgnatRange.Do(func() { mustPrefix(&cgnatRange.v, "100.64.0.0/10") })
+	cgnatRange.Do(func() { mustPrefix(&cgnatRange.v, cgnatRangeCIDR()) })
 	return cgnatRange.v
 }
 
diff --git a/ssh/tailssh/c2n.go b/ssh/tailssh/c2n.go
index 621be74..821e4c2 100644
--- a/ssh/tailssh/c2n.go
+++ b/ssh/tailssh/c2n.go
@@ -1,7 +1,7 @@
 // Copyright (c) Tailscale Inc & contributors
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build (linux && !android) || (darwin && !ios) || freebsd || openbsd || plan9
+//go:build linux || (darwin && !ios) || freebsd || openbsd || plan9
 
 package tailssh
 
diff --git a/ssh/tailssh/hostkeys.go b/ssh/tailssh/hostkeys.go
index 8046a02..358e6dd 100644
--- a/ssh/tailssh/hostkeys.go
+++ b/ssh/tailssh/hostkeys.go
@@ -1,7 +1,7 @@
 // Copyright (c) Tailscale Inc & contributors
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build (linux && !android) || (darwin && !ios) || freebsd || openbsd || plan9
+//go:build linux || (darwin && !ios) || freebsd || openbsd || plan9
 
 package tailssh
 
diff --git a/ssh/tailssh/incubator.go b/ssh/tailssh/incubator.go
index 48c65e8..ed07903 100644
--- a/ssh/tailssh/incubator.go
+++ b/ssh/tailssh/incubator.go
@@ -7,7 +7,7 @@
 // and groups to the specified `--uid`, `--gid` and `--groups`, and
 // then launches the requested `--cmd`.
 
-//go:build (linux && !android) || (darwin && !ios) || freebsd || openbsd
+//go:build linux || (darwin && !ios) || freebsd || openbsd
 
 package tailssh
 
diff --git a/ssh/tailssh/incubator_linux.go b/ssh/tailssh/incubator_linux.go
index cff4616..3539ffc 100644
--- a/ssh/tailssh/incubator_linux.go
+++ b/ssh/tailssh/incubator_linux.go
@@ -1,7 +1,7 @@
 // Copyright (c) Tailscale Inc & contributors
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build linux && !android
+//go:build linux
 
 package tailssh
 
diff --git a/ssh/tailssh/tailssh.go b/ssh/tailssh/tailssh.go
index e01f78e..72eb473 100644
--- a/ssh/tailssh/tailssh.go
+++ b/ssh/tailssh/tailssh.go
@@ -1,7 +1,7 @@
 // Copyright (c) Tailscale Inc & contributors
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build (linux && !android) || (darwin && !ios) || freebsd || openbsd || plan9
+//go:build linux || (darwin && !ios) || freebsd || openbsd || plan9
 
 // Package tailssh is an SSH server integrated into Tailscale.
 package tailssh
diff --git a/ssh/tailssh/user.go b/ssh/tailssh/user.go
index 0d2bf31..022fec6 100644
--- a/ssh/tailssh/user.go
+++ b/ssh/tailssh/user.go
@@ -1,7 +1,7 @@
 // Copyright (c) Tailscale Inc & contributors
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build (linux && !android) || (darwin && !ios) || freebsd || openbsd || plan9
+//go:build linux || (darwin && !ios) || freebsd || openbsd || plan9
 
 package tailssh
 
diff --git a/tsd/tsd.go b/tsd/tsd.go
index 615c9c0..86397f8 100644
--- a/tsd/tsd.go
+++ b/tsd/tsd.go
@@ -224,13 +224,14 @@ func (p *SubSystem[T]) Set(v T) {
 	if p.set {
 		var oldVal any = p.v
 		var newVal any = v
-		if oldVal == newVal {
-			// Allow setting to the same value.
-			// Note we had to box them through "any" to force them to be comparable.
-			// We can't set the type constraint T to be "comparable" because the interfaces
-			// aren't comparable. (See https://github.com/golang/go/issues/52531 and
-			// https://github.com/golang/go/issues/52614 for some background)
+		switch any(v).(type) {
+		case router.Router:
+			p.v = v
 			return
+		default:
+			if reflect.DeepEqual(oldVal, newVal) {
+				return
+			}
 		}
 
 		panic(fmt.Sprintf("%v is already set", reflect.TypeFor[T]().String()))
diff --git a/wgengine/router/osrouter/router_linux.go b/wgengine/router/osrouter/router_linux.go
index 41a2a7b..748dbf4 100644
--- a/wgengine/router/osrouter/router_linux.go
+++ b/wgengine/router/osrouter/router_linux.go
@@ -1,8 +1,6 @@
 // Copyright (c) Tailscale Inc & contributors
 // SPDX-License-Identifier: BSD-3-Clause
 
-//go:build !android
-
 package osrouter
 
 import (
